demos/text_detection_demo/cpp/src/cnn.cpp

```diff
@@ -119,3 +102,185 @@ InferenceEngine::BlobMap Cnn::Infer(const cv::Mat &frame) {
+    ExecutableNetwork executable_network_encoder = ie.LoadNetwork(network_encoder, deviceName);
+    ExecutableNetwork executable_network_decoder = ie.LoadNetwork(network_decoder, deviceName);
+    // ---------------------------------------------------------------------------------------------------
+
+    // --------------------------- Creating infer request ------------------------------------------------
+    infer_request_encoder_ = executable_network_encoder.CreateInferRequest();
+    infer_request_decoder_ = executable_network_decoder.CreateInferRequest();
+    // ---------------------------------------------------------------------------------------------------
+
+    is_initialized_ = true;
+}
+
+void EncoderDecoderCNN::setInOutNames(const std::string out_enc_hidden_name,
+                                    const std::string out_dec_hidden_name,
+                                    const std::string in_dec_hidden_name,
+                                    const std::string features_name,
+                                    const std::string in_dec_symbol_name,
+                                    const std::string out_dec_symbol_name,
+                                    const std::string logits_name) {
+    out_enc_hidden_name_ = out_enc_hidden_name;
+    out_dec_hidden_name_ = out_dec_hidden_name;
+    in_dec_hidden_name_ = in_dec_hidden_name;
+    features_name_ = features_name;
+    in_dec_symbol_name_ = in_dec_symbol_name;
+    out_dec_symbol_name_ = out_dec_symbol_name;
+    logits_name_ = logits_name;
+}
+
+InferenceEngine::BlobMap EncoderDecoderCNN::Infer(const cv::Mat &frame) {
+    std::chrono::steady_clock::time_point begin = std::chrono::steady_clock::now();
+
+    cv::Mat image;
+    if (channels_ == 1) {
+         cv::cvtColor(frame, image, cv::COLOR_BGR2GRAY);
+    } else {
+        image = frame.clone();
+    }
+
+    InferenceEngine::Blob::Ptr blob = infer_request_encoder_.GetBlob(input_name_);
+    matU8ToBlob<uint8_t>(image, blob);
+    // --------------------------- Doing inference -------------------------------------------------------
+    /* Running the request synchronously */
+    infer_request_encoder_.Infer();
+    // ---------------------------------------------------------------------------------------------------
+
+    // --------------------------- Processing output encoder -----------------------------------------------------
+
+    InferenceEngine::BlobMap encoder_blobs;
+    for (const auto &output_name : output_names_encoder) {
+        encoder_blobs[output_name] = infer_request_encoder_.GetBlob(output_name);
+    }
+    // blobs here are set for concrete network
+    // in case of different network this needs to be changed or generalized
+    infer_request_decoder_.SetBlob(features_name_, encoder_blobs[features_name_]);
+    infer_request_decoder_.SetBlob(in_dec_hidden_name_, encoder_blobs[out_enc_hidden_name_]);
```

Comment:
```suggestion
    // blobs here are set for concrete network
    // in case of different network this needs to be changed or generalized
    infer_request_decoder_.SetBlob(features_name_, infer_request_encoder_.GetBlob(features_name_));
    infer_request_decoder_.SetBlob(in_dec_hidden_name_, infer_request_encoder_.GetBlob(out_enc_hidden_name_));
```